thread/common/threadhelpers.cpp

```diff
@@ -70,17 +70,23 @@ omrthread_spinlock_acquire(omrthread_t self, omrthread_monitor_t monitor)
 	uintptr_t spinCount2Init = monitor->spinCount2;
 	uintptr_t spinCount1Init = monitor->spinCount1;
 
-	uintptr_t spinCount3 = spinCount3Init;
-	uintptr_t spinCount2 = spinCount2Init;
-
 #if defined(OMR_THR_SPIN_WAKE_CONTROL)
- 	if (monitor->spinThreads < lib->maxSpinThreads) {
- 		VM_AtomicSupport::add(&monitor->spinThreads, 1);
- 	} else {
- 		goto exit;
- 	}
+	BOOLEAN spinning = TRUE;
+	if (OMRTHREAD_IGNORE_SPIN_THREAD_BOUND != lib->maxSpinThreads) {
+		if (monitor->spinThreads < lib->maxSpinThreads) {
+			VM_AtomicSupport::add(&monitor->spinThreads, 1);
+		} else {
+			spinCount1Init = 1;
+			spinCount2Init = 1;
+			spinCount3Init = 1;
+			spinning = FALSE;
+		}
+	}
 #endif /* defined(OMR_THR_SPIN_WAKE_CONTROL) */
 
+	uintptr_t spinCount3 = spinCount3Init;
```

Comment:
This is effectively turning spinning off correct?

Reply:
if `spinCount*Init`(s) are set to 1, then threads will attempt one CAS operation before exiting the spin loop. this is equivalent to turning spinning off. otherwise, `spinCount*Init`retain the global values 256/32/45.